src/test/sql/sql_pipeline_test.cpp

```diff
@@ -0,0 +1,470 @@
+  EXPECT_EQ(duration, duration2);
+  EXPECT_TRUE(parsed_sql.isValid());
+
+  auto statements = parsed_sql.getStatements();
+
+  EXPECT_EQ(statements.size(), 1u);
+  EXPECT_EQ(statements.at(0)->type(), hsql::StatementType::kStmtSelect);
+}
+
+TEST_F(SQLPipelineTest, GetParsedSQLInvalid) {
+  SQLPipeline sql_pipeline{_invalid_sql};
+  EXPECT_THROW(sql_pipeline.get_parsed_sql(), std::exception);
+}
+
+TEST_F(SQLPipelineTest, GetUnoptimizedLQP) {
+  SQLPipeline sql_pipeline{_select_query_a};
+
+  const auto& lqp_roots = sql_pipeline.get_unoptimized_logical_plans();
+
+  EXPECT_EQ(lqp_roots.size(), 1u);
+  EXPECT_EQ(lqp_roots.at(0)->output_column_names(), _column_names);
+}
+
+TEST_F(SQLPipelineTest, GetUnoptimizedLQPTwice) {
+  SQLPipeline sql_pipeline{_select_query_a};
+
+  sql_pipeline.get_unoptimized_logical_plans();
+  const auto& lqp_roots = sql_pipeline.get_unoptimized_logical_plans();
+
+  EXPECT_EQ(lqp_roots.size(), 1u);
+  EXPECT_EQ(lqp_roots.at(0)->output_column_names(), _column_names);
+}
+
+TEST_F(SQLPipelineTest, GetUnoptimizedLQPWithJoinFilter) {
+  SQLPipeline sql_pipeline{_join_query};
+
+  const auto& lqp_roots = sql_pipeline.get_unoptimized_logical_plans();
+
+  EXPECT_EQ(lqp_roots.size(), 1u);
+  EXPECT_EQ(lqp_roots.at(0)->output_column_names(), _join_column_names);
+  EXPECT_TRUE(_find_in_lqp(lqp_roots.at(0), _contains_cross));
+}
+
+TEST_F(SQLPipelineTest, GetUnoptimizedLQPInvalid) {
+  SQLPipeline sql_pipeline{_invalid_sql};
+  EXPECT_THROW(sql_pipeline.get_unoptimized_logical_plans(), std::exception);
+}
+
+TEST_F(SQLPipelineTest, GetUnoptimizedLQPValidated) {
+  SQLPipeline sql_pipeline{_select_query_a};
+
+  const auto& lqp_roots = sql_pipeline.get_unoptimized_logical_plans();
+
+  EXPECT_EQ(lqp_roots.size(), 1u);
+  EXPECT_TRUE(lqp_roots.at(0)->subtree_is_validated());
+}
+
+TEST_F(SQLPipelineTest, GetUnoptimizedLQPNotValidated) {
+  SQLPipeline sql_pipeline{_select_query_a, false};
+
+  const auto& lqp_roots = sql_pipeline.get_unoptimized_logical_plans();
+
+  EXPECT_EQ(lqp_roots.size(), 1u);
+  EXPECT_FALSE(lqp_roots.at(0)->subtree_is_validated());
+}
+
+TEST_F(SQLPipelineTest, GetOptimizedLQP) {
+  SQLPipeline sql_pipeline{_select_query_a};
+
+  const auto& lqp_roots = sql_pipeline.get_optimized_logical_plans();
+
+  EXPECT_EQ(lqp_roots.size(), 1u);
+  EXPECT_EQ(lqp_roots.at(0)->output_column_names(), _column_names);
+}
+
+TEST_F(SQLPipelineTest, GetOptimizedLQPTwice) {
+  SQLPipeline sql_pipeline{_select_query_a};
+
+  sql_pipeline.get_optimized_logical_plans();
+  const auto& lqp_roots = sql_pipeline.get_optimized_logical_plans();
+
+  EXPECT_EQ(lqp_roots.size(), 1u);
+  EXPECT_EQ(lqp_roots.at(0)->output_column_names(), _column_names);
+}
+
+TEST_F(SQLPipelineTest, GetOptimizedLQPWithJoinFilter) {
+  SQLPipeline sql_pipeline{_join_query};
+
+  const auto& lqp_roots = sql_pipeline.get_optimized_logical_plans();
+
+  EXPECT_EQ(lqp_roots.size(), 1u);
+  EXPECT_FALSE(_find_in_lqp(lqp_roots.at(0), _contains_cross));
```

Comment:
You're having 2 tests with this one just adding this single line? Seems a bit overkill, I'd ditch GetOptimizedLQPWithJoinFilter and just put this line in GetOptimizedLQP. Same goes for GetUnoptimizedLQPWithJoinFilter.